Avoid an int-conversion error.  The offset variable is later used as a
pointer argument to mmap, so it makes sense to keep it as a void *
variable.

diff --git a/x86-common.c b/x86-common.c
index 137bc3ca6e12c0e1..0978276c3aa7d2c0 100644
--- a/x86-common.c
+++ b/x86-common.c
@@ -213,7 +213,7 @@ void *LRMI_common_init(int high_page)
 	if (!real_mem_init(high_page))
 		return NULL;
 
-	offset = mem_info.offset - REAL_MEM_BASE;
+	offset = (void *) (mem_info.offset - REAL_MEM_BASE);
 
 	/*
 	 Map the Interrupt Vectors (0x0 - 0x400) + BIOS data (0x400 - 0x502)
